test/test_nn.py

```diff
@@ -1621,60 +1621,68 @@ def test_ConvTranspose2d_output_size(self):
+        for tp in types:
+            m = nn.Conv2d(4, 4, kernel_size=3, groups=2).type(tp)
+            i = Variable(torch.randn(2, 4, 6, 6).type(tp), requires_grad=True)
+            output = m(i)
+            grad_output = torch.randn(2, 4, 4, 4).type(tp)
+            output.backward(grad_output)
 
-        m1 = nn.Conv2d(2, 2, kernel_size=3)
-        m1.weight.data.copy_(m.weight.data[:2])
-        m1.bias.data.copy_(m.bias.data[:2])
-        i1 = Variable(i.data[:, :2].contiguous(), requires_grad=True)
-        output1 = m1(i1)
-        output1.backward(grad_output[:, :2].contiguous())
-
-        m2 = nn.Conv2d(2, 2, kernel_size=3)
-        m2.weight.data.copy_(m.weight.data[2:])
-        m2.bias.data.copy_(m.bias.data[2:])
-        i2 = Variable(i.data[:, 2:].contiguous(), requires_grad=True)
-        output2 = m2(i2)
-        output2.backward(grad_output[:, 2:].contiguous())
-
-        self.assertEqual(output, torch.cat([output1, output2], 1))
-        self.assertEqual(i.grad.data,
-                         torch.cat([i1.grad.data, i2.grad.data], 1))
-        self.assertEqual(m.bias.grad.data,
-                         torch.cat([m1.bias.grad.data, m2.bias.grad.data], 0))
-        self.assertEqual(m.weight.grad.data,
-                         torch.cat([m1.weight.grad.data, m2.weight.grad.data], 0))
+            m1 = nn.Conv2d(2, 2, kernel_size=3).type(tp)
+            m1.weight.data.copy_(m.weight.data[:2])
+            m1.bias.data.copy_(m.bias.data[:2])
+            i1 = Variable(i.data[:, :2].contiguous(), requires_grad=True)
+            output1 = m1(i1)
+            output1.backward(grad_output[:, :2].contiguous())
+
+            m2 = nn.Conv2d(2, 2, kernel_size=3).type(tp)
+            m2.weight.data.copy_(m.weight.data[2:])
+            m2.bias.data.copy_(m.bias.data[2:])
+            i2 = Variable(i.data[:, 2:].contiguous(), requires_grad=True)
+            output2 = m2(i2)
+            output2.backward(grad_output[:, 2:].contiguous())
+
+            self.assertEqual(output, torch.cat([output1, output2], 1))
+            self.assertEqual(i.grad.data,
+                             torch.cat([i1.grad.data, i2.grad.data], 1))
+            self.assertEqual(m.bias.grad.data,
+                             torch.cat([m1.bias.grad.data, m2.bias.grad.data], 0))
+            self.assertEqual(m.weight.grad.data,
+                             torch.cat([m1.weight.grad.data, m2.weight.grad.data], 0))
 
     # For https://github.com/pytorch/pytorch/pull/1273
     # Almost identical to the above `test_Conv2d_naive_groups`
     def test_Conv2d_groups_nobias(self):
-        m = nn.Conv2d(4, 4, kernel_size=3, groups=2, bias=False)
-        i = Variable(torch.randn(2, 4, 6, 6), requires_grad=True)
-        output = m(i)
-        grad_output = torch.randn(2, 4, 4, 4)
-        output.backward(grad_output)
+        types = (torch.FloatTensor,)
+        if TEST_CUDA:
+            types += (torch.cuda.FloatTensor,)
```

Comment:
It'd be better to change this method to a helper that takes a tensor type to check and split this test into two (so that we explicitly see that a test was skipped)